ipn/serve.go

```diff
@@ -62,6 +78,41 @@ type FunnelConn struct {
 	Src netip.AddrPort
 }
 
+// ServeStreamRequest defines the json request body
+// for the serve stream endpoint
+type ServeStreamRequest struct {
+	// HostPort is the DNS and port of the tailscale
+	// URL.
+	HostPort HostPort `json:",omitempty"`
+
+	// Source is the user's serve destination
+	// such as their localhost server.
+	Source string `json:",omitempty"`
+
+	// MountPoint is the path prefix for
+	// the given HostPort.
+	MountPoint string `json:",omitempty"`
```

Comment:
"omitempty" implies this can be empty? What's its zero value mean then?

Reply:
We can treat that as an error or default to `/`? Either way is fine with me